jobbergate_cli/main.py

```diff
@@ -367,167 +573,217 @@ def create_job_script(
+@click.option("--id", "-i", "id_", help="The id of the job submission to be returned")
+@click.pass_context
+@jobbergate_command_wrapper
+def get_job_submission(ctx, id_):
     """
     GET a Job Submission.
-
-    Keyword Arguments:
-        id -- id of endpoint to action
     """
-    print(ctx.api.get_job_submission(get_job_submission_id))
+    api = ctx.obj["api"]
+    return api.get_job_submission(id_)
 
 
 @main.command("update-job-submission")
-@click.option("--id", "-i", "update_job_submission_id")
-@click.pass_obj
-@sentry_handler
-def update_job_submission(ctx, update_job_submission_id):
+@click.option("--id", "-i", "id_", help="The id of job submission to update")
+@click.pass_context
+@jobbergate_command_wrapper
+def update_job_submission(ctx, id_):
     """
     UPDATE a Job Submission.
-
-    Keyword Arguments:
-        id -- id of job submission to update
     """
-    print(ctx.api.update_job_submission(update_job_submission_id))
+    api = ctx.obj["api"]
+    return api.update_job_submission(id_)
 
 
 @main.command("delete-job-submission")
-@click.option("--id", "-i", "delete_job_submission_id")
-@click.pass_obj
-@sentry_handler
-def delete_job_submission(ctx, delete_job_submission_id):
+@click.option("--id", "-i", "id_", help="The id of job submission to delete")
+@click.pass_context
+@jobbergate_command_wrapper
+def delete_job_submission(ctx, id_):
+    """
+    DELETE a Job Submission.
     """
-    Delete a Job Submission.
+    api = ctx.obj["api"]
+    return api.delete_job_submission(id_)
+
 
-    Keyword Arguments:
-        id -- id of job submission to delete
+@main.command("upload-logs")
+@click.pass_context
+@jobbergate_command_wrapper
+def upload_logs(ctx):
+    """
+    Uploads user logs to S3 for analysis. Should only be used after an incident that was
+    reported to the Jobbergate support team.
     """
-    print(ctx.api.delete_job_submission(delete_job_submission_id))
+    logger.debug("Initializing S3 client")
+    s3_client = boto3.client(
+        "s3",
+        aws_access_key_id=JOBBERGATE_AWS_ACCESS_KEY_ID,
+        aws_secret_access_key=JOBBERGATE_AWS_SECRET_ACCESS_KEY,
+    )
+
+    tarball_name = "{user}.{timestamp}.tar.gz".format(
+        user=ctx.obj["token"]["username"],
+        timestamp=datetime.utcnow().strftime("%Y%m%d.%H%M%S"),
+    )
+
+    logger.debug("Creating tarball of user's logs")
+    log_dir = JOBBERGATE_LOG_PATH.parent
+    with tempfile.TemporaryDirectory() as temp_dir:
+        tarball_path = Path(temp_dir) / tarball_name
+        with tarfile.open(tarball_path, "w:gz") as tarball:
+            for filename in log_dir.iterdir():
+                if filename.match(f"{JOBBERGATE_LOG_PATH}*"):
+                    tarball.add(str(filename))
+
+        logger.debug(f"Uploading {tarball_name} to S3")
+        s3_client.upload_file(str(tarball_path), JOBBERGATE_S3_LOG_BUCKET, tarball_name)
+
+    return "Upload complete. Please notify Omnivector <info@omnivector.solution>."
```

Comment:
this email is correct?

Reply:
As far as I know. Double-checking with James now.

Reply:
I think it is missing an `s` at the end of `Solutions`?

Reply:
That should be fixed in the new commit.